types/node/index.d.ts

```diff
@@ -772,23 +773,29 @@ declare module "http" {
     import * as stream from "stream";
     import { URL } from "url";
 
-    export interface RequestOptions {
+    export interface ClientRequestArgs {
         protocol?: string;
         host?: string;
         hostname?: string;
         family?: number;
         port?: number | string;
+        defaultPort?: number | string;
         localAddress?: string;
         socketPath?: string;
         method?: string;
         path?: string;
         headers?: { [key: string]: any };
         auth?: string;
         agent?: Agent | boolean;
+        _defaultAgent?: Agent,
         timeout?: number;
+        rejectUnauthorized?: boolean;
+        createConnection?: Function;
     }
 
-    export interface Server extends net.Server {
+    export class Server extends net.Server {
+        constructor(requestListener?: Function);
```

Comment:
after checking source code here
https://github.com/nodejs/node/blob/master/lib/_http_server.js#L563-L593

you're right. the type should be
``` ts
constructor(requestListener?: (req: IncomingMessage, res: ServerResponse) => void);
```